subsquid/src/utils.ts

```diff
@@ -44,3 +49,43 @@ export function divideBigInts(a: bigint, b: bigint): number {
   const base = BigNumber(b.toString());
   return quote.div(base).toNumber();
 }
+
+export async function getCoingeckoPrice(assetId: "4" | "130", date?: Date): Promise<number> {
+  let time = new Date();
+  if (date && isInstance(date, Date)) {
+    time = date;
+  } else if (date) {
+    time = new Date(date);
+  }
+
+  const month = time.getMonth() + 1;
+  const day = time.getDate();
+  const year = time.getFullYear();
+
+  const queryDate = `${day < 10 ? "0" : ""}${day}-${month < 10 ? "0" : ""}${month}-${year}`;
+
+  let coinId: string;
+  switch (assetId) {
+    case "130": {
+      coinId = "tether";
+      break;
+    }
+    case "4": {
+      coinId = "kusama";
+      break;
+    }
+    default: {
+      throw new Error("Invalid assetId");
+    }
+  }
+
+  const endpoint = `https://api.coingecko.com/api/v3/coins/${coinId}/history?date=${queryDate}&localization=en`;
+  const res = await fetch(endpoint);
```

Comment:
haven't checked personally but afaik you can probably query multiple dates at once for further optimisation

Reply:
Yes this can be done but this function should be called once per asset every "day" while processing, so twice every 7,200 / 14,400 blocks. I can work on optimizing how we handle this after release.

Reply:
makes sense sir 🫡